gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/common/impl/InMemoryTopology.java

```diff
@@ -0,0 +1,140 @@
+package gobblin.config.common.impl;
+
+import gobblin.config.store.api.ConfigKeyPath;
+
+import java.util.Collection;
+import java.util.List;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.ListMultimap;
+
+/**
+ * InMemoryTopology will return stale data if the internal config store is Not {@ConfigStoreWithStableVersioning}
+ * 
+ * @author mitu
+ *
+ */
+public class InMemoryTopology implements ConfigStoreTopologyIntf {
+
+  private final ConfigStoreBackedTopology fallback;
+
+  private final ListMultimap<ConfigKeyPath, ConfigKeyPath> childrenMap = ArrayListMultimap.create();
+  private final ListMultimap<ConfigKeyPath, ConfigKeyPath> ownImportMap = ArrayListMultimap.create();
+  private final HashMultimap<ConfigKeyPath, ConfigKeyPath> ownImportedByMap = HashMultimap.create();
+  private final ListMultimap<ConfigKeyPath, ConfigKeyPath> recursiveImportMap = ArrayListMultimap.create();
+  private final HashMultimap<ConfigKeyPath, ConfigKeyPath> recursiveImportedByMap = HashMultimap.create();
+
+  public InMemoryTopology(ConfigStoreBackedTopology fallback) {
+    this.fallback = fallback;
+  }
+
+  private void loadRawTopologyFromConfigStore() {
+    // breath first search the whole topology
+    // calls to retrieve cache / set cache if not present
+    Collection<ConfigKeyPath> nodeQueue = this.getChildren(SingleLinkedListConfigKeyPath.ROOT);
+
+    while(!nodeQueue.isEmpty()){
+      ConfigKeyPath configKeyPath = nodeQueue.iterator().next();
+      nodeQueue.remove(configKeyPath);
```

Comment:
This is dangerous. You don't know if the collection returned by the getChildren() is modifiable. That's why we need a copy.